pkg/clusterclientstore/queue.go

```diff
@@ -109,12 +110,19 @@ func processNextWorkItem(wq workqueue.RateLimitingInterface, handler func(string
 
 	if key, ok = obj.(string); !ok {
 		wq.Forget(obj)
-		runtime.HandleError(fmt.Errorf("invalid object key: %#v", obj))
+		runtime.HandleError(fmt.Errorf("invalid object key (will retry: false): %#v", obj))
```

Comment:
I'm wondering if this one deserves it's own structured error too?

Reply:
I wouldn't mind turning it into one, but I also don't see the point here. `HandleError` doesn't actually *handle* anything, it just logs it as a string. In fact, it's horrible that we have to do this dance of turning it into an error first just to have it stringified again right afterward.